### PR TITLE
Unit tests for hyper block proxy

### PR DESCRIPTION
**Pull request by mariusmihaic**
- Added unit tests for `hyperBlockProxy` (100% branch coverage)
- Minor fix in `respondWithBadRequest` to respond with `CovalentHyperBlockApiResponse` instead of `shared.GenericAPIResponse`

### CODE REVIEW

File: api/hyper_block_proxy_test.go [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+		t.Parallel()
+
+		proxy, err := api.NewHyperBlockProxy(nil)
+		require.Nil(t, proxy)
+		require.Equal(t, api.ErrNilHyperBlockFacade, err)
+	})
+}
+
+func TestGetNonceFromRequest_MissingNonce_ShouldReturnError(t *testing.T) {
+	context := &gin.Context{
```
Comment (ssd04) [inline api/hyper_block_proxy_test.go]: add t.Parallel()

File: testscommon/mock/apiMocks/hyper_block_facade_stub.go [hunk 2]
```diff
@@ -0,0 +1,24 @@
+package apiMocks
+
+import "github.com/ElrondNetwork/covalent-indexer-go/api"
+
+type HyperBlockFacadeStub struct {
```
Comment (ssd04) [inline testscommon/mock/apiMocks/hyper_block_facade_stub.go]: dummy comments in this file?

Comment (bogdan-rosianu) [reply testscommon/mock/apiMocks/hyper_block_facade_stub.go]: +1

File: api/hyper_block_proxy_test.go [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+			Error: errFacade.Error(),
+			Code:  shared.ReturnCodeInternalError,
+		}, apiResp)
+	})
+}
+
+func TestHyperBlockProxy_GetHyperBlockByHash(t *testing.T) {
+	t.Parallel()
+
+	t.Run("should work", func(t *testing.T) {
```
Comment (bogdan-rosianu) [inline api/hyper_block_proxy_test.go:164]: optional: move the `should work` tests to the end of the tests?

File: api/hyper_block_proxy_test.go (hunk 1 above)
Comment (mariusmihaic) [reply api/hyper_block_proxy_test.go]: Added it

File: testscommon/mock/apiMocks/hyper_block_facade_stub.go (hunk 2 above)
Comment (mariusmihaic) [reply testscommon/mock/apiMocks/hyper_block_facade_stub.go]: Added comments

File: api/hyper_block_proxy_test.go (hunk 3 above)
Comment (mariusmihaic) [reply api/hyper_block_proxy_test.go:164]: I prefer to have the first tests with "happy path" to see the whole flow working correctly, and only after that start with error cases

**Review by ssd04 — APPROVED** [review]

**Review by bogdan-rosianu — APPROVED** [review]

**Pull request merged**